tsdb/chunks/head_chunks.go

```diff
@@ -720,17 +720,20 @@ func (cdm *ChunkDiskMapper) DeleteCorrupted(originalErr error) error {
 
 	// Delete all the head chunk files following the corrupt head chunk file.
 	segs := []int{}
+	cdm.readPathMtx.Lock()
```

Comment:
We should be holding the `RLock()` and `RUnlock()` here.

Reply:
Fixed